OlinkAnalyze/vignettes/non-parametric_analysis.Rmd

```diff
@@ -0,0 +1,230 @@
+  fig.align = "center",
+  collapse = TRUE,
+  comment = "#>")
+options(tibble.print_min = 4L, tibble.print_max = 4L)
+```
+
+```{r, echo=FALSE, message=FALSE}
+library(OlinkAnalyze)
+library(dplyr)
+library(stringr)
+library(ggplot2)
+library(kableExtra)
+```
+
+## Introduction
+
+When particular assumptions about population distribution or data characteristic are not met, a non-parametric test could be considered to analyze the data. Unlike parametric tests, which assume that the data follows a specific distribution (usually the normal distribution) and have specific parameters, non-parametric tests do not make these strict assumptions and are often referred to as distribution-free tests. Non-parametric tests are useful when dealing with ordinal, nominal, or skewed data or when the sample size is small.
+
+Here are some key characteristics and applications of non-parametric tests:
+
+No Assumptions about Data Distribution: Non-parametric tests do not assume that the data follows a specific probability distribution. This makes them robust in situations where the data’s distribution is unknown or does not meet the assumptions of parametric tests.
+
+Ordinal or Nominal Data: Non-parametric tests are well-suited for analyzing data that is categorical or ordinal, where the data points are not measured on a continuous scale. Examples include Likert scale ratings, survey responses, and categorical variables.
+
+Small Sample Sizes: Non-parametric tests are often preferred when dealing with small sample sizes because they require fewer assumptions and are less sensitive to outliers.
+
+Ranking Data: Many non-parametric tests are based on the concept of ranking data points rather than directly comparing their values. This can be particularly useful when the data is not normally distributed.
+
+Here are some common non-parametric tests:
+
+Mann-Whitney U Test (Wilcoxon Rank-Sum Test): This method is used to compare two independent groups to determine if there is a significant difference between them.
+
+Wilcoxon Signed-Rank Test: When dealing with two related (paired) samples to determine if there is a significant difference between them.
+
+Kruskal-Wallis Test: An extension of the Mann-Whitney U test used for comparing three or more independent groups.
+
+Friedman Test: An extension of the Wilcoxon Signed-Rank Test used for comparing three or more related (paired) samples.
+
+In the following examples, we will demonstrate how to perform different non-parametric tests using the dataset `npx_data1` as the reference data.
+
+First we remove the sample controls from the dataset.
+
+```{r message=FALSE, eval=T, echo = TRUE}
+npx_df <- npx_data1 %>% filter(!grepl('control',SampleID, ignore.case = TRUE))
+```
+
+## Mann-Whitney U Test
+
+Mann-Whitney U Test could be performed using the following script and specifying which column should be used as the grouping variable using the `variable` argument
+
+```{r Mann-Whitney_U_Test, echo=T, eval=T, message=FALSE}
```

Comment:
I think including all the kableExtra stuff in not necessary to show in the vignette. I would propose to have the same code as you have but with in the following way as below. That way you still have it in same format but the dont show everything. Same for all examples. 



{r Mann-Whitney_U_Test, echo=F, eval=T, message=FALSE}

and then another 
{r Mann-Whitney_U_Test, echo=T, eval=F, message=FALSE}
 
 wilcox_results <- olink_wilcox(df = npx_df,
                               variable = 'Treatment')
'# The first 10 rows of the results from Mann-Whitney U Test are shown:

wilcox_results %>% 
  head(10)